cfme/infrastructure/cluster.py

```diff
@@ -49,8 +49,8 @@ def __init__(self, name, provider, appliance=None):
         self.quad_name = 'cluster'
 
         col = rest_api().collections
-        self._id = [cl.id for cl in col.clusters.all if cl.name == self._short_name
-                    and cl.ems_id == self.provider.id][-1]
+        self._id = [cl.id for cl in col.clusters.all if cl.name == self._short_name and
```

Comment:
a more nice formatting for multiline conditional list expression is newline before expression * for and another newline before the condition, so one line will be the loop, and another line will be the condition

btw, why do we obtain the last element instead of the first matching?

if it its ok to take the first one, we could use next + a generator expression